### PR TITLE
API: Handle pow & rpow special cases

### LINKED ISSUE #29997: pow(1, pd.NA) maybe gives the wrong result
Should `pow(1, pd.NA)` be `1` or `NA`?

```python
In [1]: import pandas as pd

In [2]: 1 ** pd.NA
Out[2]: NA

In [3]: import numpy as np

In [5]: 1 ** np.nan
Out[5]: 1.0
```

cc @jorisvandenbossche

### PR DESCRIPTION
**Pull request by TomAugspurger**
Closes https://github.com/pandas-dev/pandas/issues/29997

@jorisvandenbossche do you have thoughts on where to document this behavior? I added a bit to `reference/arrays.rst` for now, but perhaps we'll eventually have a `user_guide/scalar_na.rst`?

### CODE REVIEW

File: pandas/tests/scalar/test_na_scalar.py [hunk 1]
```diff
@@ -69,6 +72,25 @@ def test_comparison_ops():
...
+    assert result == 0
+
+
+@pytest.mark.parametrize(
+    "value", [1, 1.0, True, np.bool_(True), np.int_(1), np.float_(1)]
+)
+def test_rpow_special(value):
+    result = value ** pd.NA
+    assert result == 1
+    if not isinstance(value, (np.float_, np.bool_, np.int_)):
```
Comment (TomAugspurger) [inline pandas/tests/scalar/test_na_scalar.py]: Somebody (Cython?) is converting these to Python scalars.

Comment (jbrockmendel) [reply pandas/tests/scalar/test_na_scalar.py]: have you tried with/without numexpr?

**Comment by jorisvandenbossche** [comment]
> @jorisvandenbossche do you have thoughts on where to document this behavior? 

The NA scalar itself is currently documented in missing_data.rst. I would maybe add it there?

File: pandas/tests/scalar/test_na_scalar.py (hunk 1 above)
Comment (jorisvandenbossche) [reply pandas/tests/scalar/test_na_scalar.py]: How can numexpr be involved in this?

**Comment by TomAugspurger** [comment]
Didn't see that section. Moved the note there.

We will still need something in reference so that :class:\`NA\` points somewhere.

**Comment by TomAugspurger** [comment]
This is blocking https://github.com/pandas-dev/pandas/pull/29964, so it'd be nice to merge this in soon if possible.

File: doc/source/user_guide/missing_data.rst [hunk 2]
```diff
@@ -832,6 +832,12 @@ return ``False``.
    pd.NA == pd.NA
    pd.NA < 2.5
 
+There are a few special cases when the result is known, even when one of the
+operands is ``NA``.
+
+* ``pd.NA ** 0`` is always 0.
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/missing_data.rst]:
```suggestion
* ``pd.NA ** 0`` is always 1.
```

File: pandas/_libs/missing.pyx [hunk 3]
```diff
@@ -383,6 +381,29 @@ class NAType(C_NAType):
     __abs__ = _create_unary_propagating_op("__abs__")
     __invert__ = _create_unary_propagating_op("__invert__")
 
+    # pow has special
+    def __pow__(self, other):
+        if other is C_NA:
+            return NA
+        elif isinstance(other, (numbers.Number, np.bool_)):
+            if other == 0:
+                return other
```
Comment (jorisvandenbossche) [inline pandas/_libs/missing.pyx]: This should actually be 1 right?
But, the question is then if this should be int or float, and I suppose it should be of the same type as other?

File: doc/source/user_guide/missing_data.rst [hunk 4]
```diff
@@ -832,6 +832,12 @@ return ``False``.
    pd.NA == pd.NA
    pd.NA < 2.5
 
+There are a few special cases when the result is known, even when one of the
```
Comment (jorisvandenbossche) [inline doc/source/user_guide/missing_data.rst]: I would maybe move this above the equality/comparison operations (as this belongs to the arithmetic operations, and the isna(..) also follows more closely on comparison operations then)

File: doc/source/user_guide/missing_data.rst (hunk 2 above)
Comment (TomAugspurger) [reply doc/source/user_guide/missing_data.rst]: Whoops, yes.

File: pandas/_libs/missing.pyx (hunk 3 above)
Comment (TomAugspurger) [reply pandas/_libs/missing.pyx]: Making it the same type as other is probably best.

**Comment by TomAugspurger** [comment]
Addressed those, thanks for catching that.

File: pandas/tests/scalar/test_na_scalar.py (hunk 1 above)
Comment (jbrockmendel) [reply pandas/tests/scalar/test_na_scalar.py]: i dont know, but if we're brainstorming things that can do surprising conversions, it comes to minid

File: pandas/tests/scalar/test_na_scalar.py [hunk 5]
```diff
@@ -69,6 +72,25 @@ def test_comparison_ops():
...
+)
+def test_pow_special(value):
+    result = pd.NA ** value
+    assert isinstance(result, type(value))
+    assert result == 1
+
+
+@pytest.mark.parametrize(
+    "value", [1, 1.0, True, np.bool_(True), np.int_(1), np.float_(1)]
+)
```
Comment (jbrockmendel) [inline pandas/tests/scalar/test_na_scalar.py]: corner case to check -0.0

Comment (TomAugspurger) [reply pandas/tests/scalar/test_na_scalar.py]: Good catch.

```python
In [4]: -1 ** np.nan
Out[4]: -1.0

In [5]: np.nan ** -0
Out[5]: 1.0

```

Will match the NumPy behavior here.

File: pandas/tests/scalar/test_na_scalar.py [hunk 6]
```diff
@@ -69,6 +72,49 @@ def test_comparison_ops():
         assert (other <= NA) is NA
 
 
+@pytest.mark.parametrize(
```
Comment (WillAyd) [inline pandas/tests/scalar/test_na_scalar.py]: There are fixtures defined for zero / one values in pandas/tests/indexing/conftest.py - can these be combined with that?

**Review by WillAyd — CHANGES_REQUESTED** [review]
General question - is this behavior defined or standardized somewhere?

File: pandas/tests/scalar/test_na_scalar.py (hunk 6 above)
Comment (TomAugspurger) [reply pandas/tests/scalar/test_na_scalar.py]: That seems to include arrays. We just want a scalar here I think.

**Comment by TomAugspurger** [comment]
> General question - is this behavior defined or standardized somewhere?

It's part of some IEEE spec, but I haven't tracked down the source. Both Python and NumPy use this behavior for NaN.

**Review by WillAyd — COMMENTED** [review]
Thanks for clarifying comments. Good on all points from my perspective

**Comment by jreback** [comment]
thanks @TomAugspurger

**Pull request merged**